File: python/web_pytest/web_playwright/pages/ajax_page.py
from playwright.sync_api import Page, expect


class AjaxPage:

    def __init__(self, page: Page):
        self.page = page
        self.button = page.locator("#button")
        self.title = page.locator("#title")

    def click(self):
        self.button.click()

    def verify_text(self):
        expect(self.title).to_be_visible(timeout=15_000)
        expect(self.title).to_have_text("AJAX")

File: nodejs/playwright/pages/ajax.page.ts
import { expect, type Locator, type Page } from '@playwright/test';


class AjaxPage {

  page: Page;
  button: Locator;
  title: Locator;

  constructor(page: Page) {
    this.page = page;
    this.button = page.locator("#button");
    this.title = page.locator("#title");
  }


  async click() {
    await this.button.click();
  }

  async verify() {
    await expect(this.title).toBeVisible();
    await expect(this.title).toHaveText("AJAX");
  }

}

export default AjaxPage;
